src/app/scripts/cac/map/cac-map-control.js

```diff
@@ -433,49 +433,72 @@ CAC.Map.Control = (function ($, Handlebars, cartodb, L, turf, _) {
+                // The marker closes with a timeout as well, and we want to hold on to it so we can
+                // cancel it if we want to
+                var markerTimeout;
+
+                lastItineraryHoverMarker = new cartodb.L.circleMarker(e.latlng, {
+                    color: '#40a368',
+                    fillOpacity: 0.5,
+                    radius: 8,
+                });
+
+                var closePopup = function() {
+                    if (lastItineraryHoverMarker) {
+                        lastItineraryHoverMarker.closePopup();
+                    }
+                };
 
                 lastItineraryHoverMarker.bindPopup('Drag marker to change route',
-                                                   {closeButton: false})
+                                                   {closeButton: false, offset: L.point(0, -7) })
                     .on('mouseover', function() {
                         clearTimeout(popupTimeout);
+                        clearTimeout(markerTimeout);
                         return dragging || this.openPopup();
                     }).on('mouseout', function() {
                         // Close popup, but with a slight delay to avoid flickering, and with an
                         // existence check to avoid errors if the marker has been destroyed
-                        popupTimeout = setTimeout(function() {
-                            if (lastItineraryHoverMarker) {
-                                lastItineraryHoverMarker.closePopup();
-                            }
-                         }, 50);
+                        popupTimeout = setTimeout(closePopup, 50);
 
                         // hide marker after awhile if not dragging
-                        if (dragging) {
-                            return;
+                        if (!dragging) {
+                            markerTimeout = setTimeout(function() {
+                                if (lastItineraryHoverMarker && !dragging) {
+                                    map.removeLayer(lastItineraryHoverMarker);
+                                    lastItineraryHoverMarker = null;
+                                    dragging = false;
+                                    startDragPoint = null;
+                                }
+                            }, 3000);
                         }
-
-                        setTimeout(function() {
-                            if (lastItineraryHoverMarker && !dragging) {
-                                map.removeLayer(lastItineraryHoverMarker);
-                                lastItineraryHoverMarker = null;
-                                dragging = false;
-                                startDragPoint = null;
-                            }
-                        }, 3000);
                     });
 
+                // circleMarkers aren't draggable, so this implments dragging using mouse
+                // down/move/up events. What could go wrong?
```

Comment:
:laughing: 

The reason why I'd suggested a preference toward making a circle icon for the already-draggable marker is that we wouldn't have to deal with anything broken with a custom draggable implementation, or any weird browser edge cases.